src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -265,43 +258,51 @@ const useGroupInfo = () => {
       user_id: item,
       channel_role: 'channel_moderator'
     }));
+    const user = {
+      id: profile?.myProfile?.user_id
+    };
+    const token = TokenStorage.get(ITokenEnum.token);
     await setOpenModal(false);
-    const filterMessage = await client.client.queryChannels(filter, sort, {
-      watch: true, // this is the default
-      state: true
-    });
-    navigation.reset({
-      index: 1,
-      routes: [
-        {
-          name: 'AuthenticatedStack',
-          params: {
-            screen: 'HomeTabs',
+    try {
+      await client.client.connectUser(user, token);
+      const filterMessage = await client.client.queryChannels(filter, sort, {
+        watch: true, // this is the default
+        state: true
+      });
+      navigation.reset({
+        index: 1,
+        routes: [
+          {
+            name: 'AuthenticatedStack',
             params: {
-              screen: 'ChannelList'
+              screen: 'HomeTabs',
+              params: {
+                screen: 'ChannelList'
+              }
+            }
+          },
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen: 'ChatDetailPage'
             }
           }
-        },
-        {
-          name: 'AuthenticatedStack',
-          params: {
-            screen: 'ChatDetailPage'
-          }
-        }
-      ]
-    });
-    const generatedChannelId = generateRandomId();
-
-    if (filterMessage.length > 0) {
-      setChannel(filterMessage[0], dispatchChannel);
-    } else {
-      const channelChat = await client.client.channel('messaging', generatedChannelId, {
-        name: selectedUser.user.name,
-        type_channel: 1
+        ]
       });
-      await channelChat.create();
-      await channelChat.addMembers(memberWithRoles);
-      setChannel(channelChat, dispatchChannel);
+      const generatedChannelId = generateRandomId();
+      if (filterMessage.length > 0) {
+        setChannel(filterMessage[0], dispatchChannel);
+      } else {
+        const channelChat = await client.client.channel('messaging', generatedChannelId, {
+          name: selectedUser.user.name,
+          type_channel: 1
+        });
+        await channelChat.create();
+        await channelChat.addMembers(memberWithRoles);
+        setChannel(channelChat, dispatchChannel);
+      }
+    } catch (e) {
+      console.log({e, client}, 'error');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section has a lot of changes related to the chat creation process. Make sure that these changes are thoroughly tested, especially the parts where you're interacting with the `client.client` object and manipulating the navigation stack.



<!-- This is an auto-generated comment by OSS CodeRabbit -->